zanata-model/src/main/java/org/zanata/model/Request.java

```diff
@@ -0,0 +1,126 @@
+ * the License, or (at your option) any later version.
+ *
+ * This software is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE. See the GNU
+ * Lesser General Public License for more details.
+ *
+ * You should have received a copy of the GNU Lesser General Public
+ * License along with this software; if not, write to the Free
+ * Software Foundation, Inc., 51 Franklin St, Fifth Floor, Boston, MA
+ * 02110-1301 USA, or see the FSF site: http://www.fsf.org.
+ */
+
+package org.zanata.model;
+
+import lombok.AccessLevel;
+import lombok.Getter;
+import lombok.NoArgsConstructor;
+import lombok.Setter;
+import org.hibernate.annotations.Type;
+import org.hibernate.annotations.TypeDef;
+import org.hibernate.annotations.TypeDefs;
+import org.zanata.model.type.RequestState;
+import org.zanata.model.type.RequestStateType;
+import org.zanata.model.type.RequestType;
+import org.zanata.model.type.RequestTypeType;
+
+import javax.persistence.Access;
+import javax.persistence.AccessType;
+import javax.persistence.Column;
+import javax.persistence.Entity;
+import javax.persistence.EntityListeners;
+import javax.persistence.JoinColumn;
+import javax.persistence.ManyToOne;
+import javax.persistence.PrePersist;
+import javax.validation.constraints.NotNull;
+import javax.validation.constraints.Size;
+import java.util.Date;
+
+/**
+ * @author Alex Eng <a href="aeng@redhat.com">aeng@redhat.com</a>
+ */
+@Access(AccessType.FIELD)
+@Getter
+@Setter(AccessLevel.PRIVATE)
+@Entity
+@EntityListeners({ Request.EntityListener.class })
+@TypeDefs({
+    @TypeDef(name = "requestState", typeClass = RequestStateType.class),
+    @TypeDef(name = "requestType", typeClass = RequestTypeType.class)
+
+})
+@NoArgsConstructor
+public class Request extends TimeEntityBase {
+
+    @Type(type = "requestState")
+    @Column(nullable = true)
+    private RequestState state;
+
+    @Type(type = "requestType")
+    @Column(nullable = false)
+    @NotNull
+    private RequestType requestType;
+
+    @Column(nullable = true)
+    @Size(max = 255)
+    private String comment;
+
+    //requesting account.
+    @ManyToOne
+    @JoinColumn(name = "requesterId", nullable = false)
+    @NotNull
+    private HAccount requester;
+
+    //account who actioned on the request.
+    @ManyToOne
+    @JoinColumn(name = "actorId", nullable = true)
+    private HAccount actor;
+
+    public Request(RequestType requestType, HAccount requester,
+        String entityId) {
+        this.requestType = requestType;
+        this.requester = requester;
+        this.entityId = entityId;
+    }
+
+    /**
+     * Return new request with updated state.
+     * Expire this request (set validTo)
+     *
+     * @param actor
+     * @param state
+     * @param comment
+     */
+    public Request update(HAccount actor, RequestState state, String comment) {
```

Comment:
This is a bit like a `with` method (because it returns a clone with different field values).  Except for the important fact that it also expires the old record.

`update` is probably fine, but we could also consider `updateWith`, `replaceWith`, `expireWithReplace`...